src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,74 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ELEMENTS = 10;
+    private static final int MIN_INDEX = 0;
+    private KeyValue<K, V>[] keysValues;
+    private int size;
+
+    public StorageImpl() {
+        keysValues = new KeyValue[MAX_ELEMENTS];
+    }
+
     @Override
     public void put(K key, V value) {
+        int keyIndex = getKeyIndex(key);
+        if (keyIndex >= MIN_INDEX) {
+            keysValues[keyIndex].setValue(value);
+        } else {
+            try {
+                keysValues[size++] = new KeyValue<>(key, value);
+            } catch (RuntimeException e) {
+                throw new RuntimeException("You can't put at the storage more than "
+                        + MAX_ELEMENTS + " elements");
+            }
```

Comment:
```suggestion
        if (keyIndex >= MIN_INDEX) {
            keysValues[keyIndex].setValue(value);
            return;
        } 
        if (size >= MAX_ELEMENTS) {
        throw new RuntimeException("You can't put at the storage more than "
                        + MAX_ELEMENTS + " elements");
       }
      keysValues[size++] = new KeyValue<>(key, value);
```
1) It’s bad practice to complicate your code if-else constructions
2) It's a bad practice catch RuntimeExceptions